src/mongoose_account_api.erl

```diff
@@ -0,0 +1,327 @@
+    Users = ejabberd_auth:get_vh_registered_users(Domain),
+    DeletedUsers = delete_old_users(Days, Users),
+    lists:map(fun jid:to_binary/1, DeletedUsers).
+
+-spec delete_old_users(Days, Users) -> Users when Days :: integer(),
+                                                  Users :: [jid:simple_bare_jid()].
+delete_old_users(Days, Users) ->
+    %% Convert older time
+    SecOlder = Days*24*60*60,
+    %% Get current time
+    TimeStampNow = erlang:system_time(second),
+    %% Apply the remove function to every user in the list
+    lists:filter(fun(User) ->
+                         delete_old_user(User, TimeStampNow, SecOlder)
+                 end, Users).
+
+format_deleted_users(Users) ->
+    io_lib:format("Deleted ~p users: ~p", [length(Users), Users]).
+
+-spec delete_old_user(User :: jid:simple_bare_jid(),
+                      TimeStampNow :: non_neg_integer(),
+                      SecOlder :: non_neg_integer()) -> boolean().
+delete_old_user({LUser, LServer}, TimeStampNow, SecOlder) ->
+    %% Check if the user is logged
+    JID = jid:make(LUser, LServer, <<>>),
+    case ejabberd_sm:get_user_resources(JID) of
+        [] -> delete_old_user_if_nonactive_long_enough(JID, TimeStampNow, SecOlder);
+        _ -> false
+    end.
+
+-spec delete_old_user_if_nonactive_long_enough(jid:jid(), non_neg_integer(),
+                                               non_neg_integer()) ->boolean().
+delete_old_user_if_nonactive_long_enough(JID, TimeStampNow, SecOlder) ->
+    case is_user_nonactive_long_enough(JID, TimeStampNow, SecOlder) of
+        true ->
+            ejabberd_auth:remove_user(JID),
+            true;
+        false ->
+            false
+    end.
+
+-spec is_user_nonactive_long_enough(jid:jid(), non_neg_integer(), non_neg_integer()) -> boolean().
+is_user_nonactive_long_enough(JID, TimeStampNow, SecOlder) ->
+    {LUser, LServer} = jid:to_lus(JID),
+    {ok, HostType} = mongoose_domain_api:get_domain_host_type(LServer),
+    case mod_last:get_last_info(HostType, LUser, LServer) of
+        {ok, TimeStamp, _Status} ->
+            % Get user age
+            Sec = TimeStampNow - TimeStamp,
+            case Sec < SecOlder of
+                % Younger than SecOlder
+                true ->
+                    false;
+                %% Older
+                false ->
+                    true
+            end;
+        not_found ->
+            true
+    end.
+
+format_change_password(ok) ->
+    {ok, "Password changed"};
+format_change_password({error, empty_password}) ->
+    {empty_password, "Empty password"};
+format_change_password({error, not_allowed}) ->
+    {not_allowed, "Password change not allowed"};
+format_change_password({error, invalid_jid}) ->
+    {invalid_jid, "Invalid jid"}.
+
+-spec kick_sessions(jid:jid(), binary()) -> [ok].
+kick_sessions(JID, Reason) ->
+    lists:map(
+        fun(Resource) ->
+                service_admin_extra_sessions:kick_session(
```

Comment:
This needs to be moved to `mongoose_session_api` - otherwise I think we may forget about such dependencies that we need to avoid.